socket_utility.c

```diff
@@ -64,13 +66,24 @@ int32_t socket_utility_create_unix_socket(int32_t type, int32_t protocol,
 		return -1;
 	}
 
+	if (group_id)
+            socket_mode = S_IRWXU | S_IRGRP | S_IWGRP;
+        else
+            socket_mode = S_IRWXU;
+
```

Comment:
This is not needed. You can just set S_IRWXU | S_IRGRP | S_IWGRP, it will be the same (group is root?).